fuel-vm/src/tests/blockchain.rs

```diff
@@ -1506,3 +1506,131 @@ fn timestamp_works() {
         }
     }
 }
+
+#[rstest::rstest]
+fn block_height_works(#[values(0, 1, 2, 10, 100)] current_height: u32) {
+    let current_height: BlockHeight = current_height.into();
+
+    let mut client = MemoryClient::default();
+
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = Default::default();
+
+    let params = *client.params();
+
+    client.as_mut().set_block_height(current_height);
+
+    #[rustfmt::skip]
+    let script = vec![
+        op::bhei(0x20),         // perform the instruction
+        op::log(0x20, 0, 0, 0), // log output
+        op::ret(RegId::ONE)
+    ];
+
+    let script = script.into_iter().collect();
+    let script_data = vec![];
+
+    let tx = TransactionBuilder::script(script, script_data)
+        .gas_price(gas_price)
+        .gas_limit(gas_limit)
+        .maturity(maturity)
+        .with_params(params)
+        .finalize_checked(current_height, client.gas_costs());
+
+    let receipts = client.transact(tx);
+    let Some(Receipt::Log { ra, .. }) = receipts.first() else {
+        panic!("expected log receipt");
+    };
+
+    let r: u32 = (*ra).try_into().unwrap();
+    let result: BlockHeight = r.into();
+    assert_eq!(result, current_height);
+}
+
+#[rstest::rstest]
+fn block_hash_works(#[values(0, 1, 2, 10, 100)] current_height: u32, #[values(0, 1, 2, 10, 100)] test_height: u32) {
+    let current_height: BlockHeight = current_height.into();
+    let test_height: BlockHeight = test_height.into();
+
+    let mut client = MemoryClient::default();
+
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = Default::default();
+
+    let params = *client.params();
+
+    client.as_mut().set_block_height(current_height);
+
+    let expected = client
+        .as_ref()
+        .block_hash(test_height)
+        .expect("failed to calculate block hash");
+
+    #[rustfmt::skip]
+    let script = vec![
+        op::movi(0x10, 32),                 // allocation size
+        op::aloc(0x10),                     // allocate memory
+        op::movi(0x11, test_height.into()), // set the argument
+        op::bhsh(RegId::HP, 0x11),          // perform the instruction
+        op::logd(0, 0, RegId::HP, 0x10),    // log output
+        op::ret(RegId::ONE)
+    ];
+
+    let script = script.into_iter().collect();
+    let script_data = vec![];
+
+    let tx = TransactionBuilder::script(script, script_data)
+        .gas_price(gas_price)
+        .gas_limit(gas_limit)
+        .maturity(maturity)
+        .with_params(params)
+        .finalize_checked(current_height, client.gas_costs());
+
+    let receipts = client.transact(tx);
+    let Some(Receipt::LogData { data, .. }) = receipts.first() else {
+        panic!("expected log receipt");
+    };
+
+    assert_eq!(data, &*expected);
+}
+
+#[rstest::rstest]
+fn block_proposer_works() {
```

Comment:
This should really be coinbase, as the proposer can use a different wallet for collecting fees (i.e. cold storage)

Reply:
So is the spec wrong in this case? https://github.com/FuelLabs/fuel-specs/blob/master/src/vm/instruction_set.md#cb-block-proposer-address

Reply:
if this will always be the coinbase, then is seems like the short text of the opcode needs to be changed

Reply:
Yeah there's not a lot of benefit to having the block proposer address, John and I previously discussed that this should just be the coinbase address associated with the coinbase transaction on the block. We could add another opcode for block proposer later if we want to.

Reply:
Made a PR for the specs here: https://github.com/FuelLabs/fuel-specs/pull/477

Reply:
Updated PR